client/src/components/Filter.jsx

```diff
@@ -10,8 +24,208 @@ import React from "react";
  *
  * Cancel should clear any unapplied filters and close the dropdown
  */
-const Filter = () => {
-  return <></>;
+
+const Filter = ({ filterLabel }) => {
+  const [allFilterData, setAllFilterData] = useContext(FilterContext);
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [icon, setIcon] = useState(<ArrowDropDown />);
+  const [buttonBgColor, setButtonBgColor] = useState("blue.200");
+  const [buttonBorderColor, setButtonBorderColor] = useState("gray.50");
+
+  const [checked, setChecked] = useState(allFilterData[filterLabel].data);
+  const [cancelButton, setCancelButton] = useState("");
```

Comment:
left some comments about state here, lmk if you need to discuss anything more, we have quite a bit of state that might be more than we need here